Cython/Utility/arrayarray.h

```diff
@@ -105,9 +106,20 @@ PyObject* newarrayobject(PyTypeObject *type, Py_ssize_t size,
     struct arraydescr *descr);
 #endif /* ifndef NO_NEWARRAY_INLINE */
 
+static CYTHON_INLINE __data_union __Pyx_PyArray_Data(arrayobject *self) {
+#if !CYTHON_COMPILING_IN_GRAAL
+    return self->data;
+#else
+    __data_union data;
+    data.ob_item = GraalPyArray_Data((PyObject*)self);
+    return data;
+#endif
```

Comment:
```suggestion
#if CYTHON_COMPILING_IN_GRAAL
    __Pyx_data_union data;
    data.ob_item = GraalPyArray_Data((PyObject*)self);
    return data;
#else
    return self->data;
#endif
```

We've generally tried to leave the general case in the `else` just because it tends to be easier to write if the list gets longer.